### PR TITLE
docs(UPGRADE.md): add section about using MeshHTTPRoute in GAPI

### PR DESCRIPTION
**Pull request by bartsmykla**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - No relevant issues
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - It won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - No tests necessary
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label)
  - It doesn't

> Changelog: skip

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -126,6 +126,27 @@ In the upcoming release, Kuma will redirect IPv6 traffic to the same port as IPv
...
 We recommend that you update your configurations to use the new defaults for IPv6 traffic redirection. If you need to retain separate ports for IPv4 and IPv6 traffic, you can continue to use the deprecated flags and configuration options until they are removed.
 
+### Change in internal resources with Kubernetes Gateway API
+
+This section describes changes to internal resources used by Kuma when configuring the built-in gateway using the Kubernetes Gateway API.
+
+#### Prior Behavior (Before Kuma 2.7.0):
+
+  * Applying a `Gateway` resource resulted in the creation of corresponding `MeshGateway` and `MeshGatewayInstance` resources.
+  * Applying an `HTTPRoute` resource was converted to a `MeshGatewayRoute` resource.
```
Comment (slonka) [inline UPGRADE.md:136]: not a native speaker but, shouldn't this be like?

```suggestion
  * An applied `HTTPRoute` resource was converted to a `MeshGatewayRoute` resource.
```

File: UPGRADE.md [hunk 2]
```diff
@@ -126,6 +126,27 @@ In the upcoming release, Kuma will redirect IPv6 traffic to the same port as IPv
...
+  * `HTTPRoute` resources are now converted to `MeshHTTPRoute` resources instead of `MeshGatewayRoute` resources.
+
+#### Upgrade Impact:
+
+  * Existing `MeshGatewayRoute` resources automatically created from `HTTPRoute` definitions will be deleted during the upgrade.
+  * New `MeshHTTPRoute` resources will be created to replace the deleted ones.
+
+#### Important Note:
+
+This change is transparent with regard to the generated Envoy configuration. There should be no impact on existing traffic routing.
```
Comment (slonka) [inline UPGRADE.md:149]: shouldn't we start with this? that there should be no impact to the live traffic?

Comment (bartsmykla) [reply UPGRADE.md:149]: I don't think so. This information may seem extensive, but it's always worth to read it fully. While I've tried to anticipate all potential impacts of this change, there's always a chance something unexpected could arise. Reading this entire section will ensure you're fully informed of the changes and any potential consequences, even those I may not have foreseen.

Comment (michaelbeaumont) [reply UPGRADE.md:149]: Yeah I agree, ultimately `UPGRADE.md` just isn't that long, we can try to push users to read it all.

**Review by Automaat — APPROVED** [review]

**Pull request merged**